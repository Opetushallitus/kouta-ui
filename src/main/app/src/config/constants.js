export const ENV_MOCK = 'ENV_MOCK';
export const ENV_DEV = 'ENV_DEV';
export const LANGUAGE = 'FI';

export const SERVICE_DOMAINS =  {
  [ENV_DEV]: "https://virkailija.hahtuvaopintopolku.fi",
  [ENV_MOCK]: "http://localhost:3001"
};

export const JULKAISUTILA = {
  TALLENNETTU: 'tallennettu',
  JULKAISTU: 'julkaistu'
};

export const REQUEST_STATUS = {
  SUCCESS: 'success',
  FAILURE: 'failure',
  DISABLED: 'disabled',
  ENABLED: 'enabled'
};

export const KOULUTUSTYYPPI_CATEGORY = {
  AMMATILLINEN_KOULUTUS: 'amm',
  KORKEAKOULUKOULUTUS: 'kk',
  LUKIOKOULUTUS: 'lk',
  MUU_KOULUTUS: 'muu'
};

export const ENTITY_MODIFICATION_MODE = {
  CREATE_ENTITY: 'CREATE_ENTITY',
  INHERIT_ENTITY: 'INHERIT_ENTITY',
  USE_ENTITY: 'USE_ENTITY'
}

export const WORKFLOW = {
  KOULUTUS: 'WORKFLOW_KOULUTUS',
  TOTEUTUS: 'WORKFLOW_TOTEUTUS',
  HAKU: 'WORKFLOW_HAKU',
  HAKUKOHDE: 'WORKFLOW_HAKUKOHDE',
  VALINTAPERUSTEET: 'WORKFLOW_VALINTAPERUSTEET'
}

export const KOULUTUSTYYPPI_CATEGORY_TO_KOULUTUSTYYPPI_IDS_MAP = {
  [KOULUTUSTYYPPI_CATEGORY.AMMATILLINEN_KOULUTUS]: [
    'koulutustyyppi_1',
    'koulutustyyppi_4',
    'koulutustyyppi_5',
    'koulutustyyppi_7',
    'koulutustyyppi_8',
    'koulutustyyppi_11',
    'koulutustyyppi_12',
    'koulutustyyppi_13',
    'koulutustyyppi_18',
    'koulutustyyppi_19',
    'koulutustyyppi_26',
    'koulutustyyppi_26'
  ],
  [KOULUTUSTYYPPI_CATEGORY.KORKEAKOULUKOULUTUS]: [
    'koulutustyyppi_3'
  ],
  [KOULUTUSTYYPPI_CATEGORY.LUKIOKOULUTUS]: [
    'koulutustyyppi_2',
    'koulutustyyppi_9',
    'koulutustyyppi_14',
    'koulutustyyppi_23'
  ],
  [KOULUTUSTYYPPI_CATEGORY.MUU_KOULUTUS]: [
    'koulutustyyppi_6',
    'koulutustyyppi_10',
    'koulutustyyppi_15',
    'koulutustyyppi_16',
    'koulutustyyppi_24',
    'koulutustyyppi_21',
    'koulutustyyppi_22'
  ]
};

export const LANGUAGE_CODE_TO_LANGUAGE_NAME = {
  'fi': 'Suomi',
  'sv': 'Ruotsi',
  'en': 'Englanti'
}

export const LANGUAGE_CODE_TO_TAB_NAME = {
  'fi': 'Suomeksi',
  'sv': 'Ruotsiksi',
  'en': 'Englanniksi'
};
