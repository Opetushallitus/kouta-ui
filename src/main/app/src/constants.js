export const LANGUAGE = 'fi';

export const JULKAISUTILA = {
  TALLENNETTU: 'tallennettu',
  JULKAISTU: 'julkaistu',
  ARKISTOITU: 'arkistoitu',
};

export const KOULUTUSTYYPPI = {
  AMMATILLINEN_KOULUTUS: 'amm',
  LUKIOKOULUTUS: 'lk',
  YLIOPISTOKOULUTUS: 'yo',
  AMKKOULUTUS: 'amk',
  VALMA: 'valma',
  TELMA: 'telma',
  TUTKINNON_OSA: 'tutkinnon_osa',
  OSAAMISALA: 'osaamisala',
  AVOIN_YO: 'avoin_yo',
  AVOIN_AMK: 'avoin_amk',
  TAYDENNYS_KOULUTUS: 'taydennyskoulutus',
  ERIKOISTUMISKOULUTUS: 'erikoistumiskoulutus',
  VALMENTAVA_KOULUTUS: 'valmentava_koulutus',
  AMMATILLINEN_OPETTAJAKOULUTUS: 'ammatillinen_opettajakoulutus',
  VAPAA_SIVISTYSTYO: 'vapaa_sivistystyo',
  LUVA: 'luva',
  MUUT_KOULUTUKSET: 'muut_koulutukset',
  PERUSOPETUKSEN_LISAOPETUS: 'perusopetuksen_lisaopetus',
};

export const TUTKINTOON_JOHTAVAT_KORKEAKOULU_KOULUTUSTYYPIT = [
  KOULUTUSTYYPPI.AMKKOULUTUS,
  KOULUTUSTYYPPI.YLIOPISTOKOULUTUS,
];

export const TUTKINTOON_JOHTAVAT_KOULUTUSTYYPIT = [
  KOULUTUSTYYPPI.AMMATILLINEN_KOULUTUS,
  KOULUTUSTYYPPI.LUKIOKOULUTUS,
  ...TUTKINTOON_JOHTAVAT_KORKEAKOULU_KOULUTUSTYYPIT,
];

export const TUTKINTOON_JOHTAVAT_AMMATILLISET_KOULUTUSTYYPIT = [
  KOULUTUSTYYPPI.AMMATILLINEN_KOULUTUS,
];

export const TUTKINTOON_JOHTAMATON_KOULUTUSTYYPPIHIERARKIA = [
  {
    tyyppi: 'ammatillinen',
    children: [
      { tyyppi: KOULUTUSTYYPPI.VALMA },
      { tyyppi: KOULUTUSTYYPPI.TELMA },
      { tyyppi: KOULUTUSTYYPPI.TUTKINNON_OSA },
      { tyyppi: KOULUTUSTYYPPI.OSAAMISALA },
    ],
  },
  {
    tyyppi: 'korkeakoulutus',
    children: [
      { tyyppi: KOULUTUSTYYPPI.AVOIN_YO },
      { tyyppi: KOULUTUSTYYPPI.AVOIN_AMK },
      { tyyppi: KOULUTUSTYYPPI.TAYDENNYS_KOULUTUS },
      { tyyppi: KOULUTUSTYYPPI.ERIKOISTUMISKOULUTUS },
      { tyyppi: KOULUTUSTYYPPI.VALMENTAVA_KOULUTUS },
      { tyyppi: KOULUTUSTYYPPI.AMMATILLINEN_OPETTAJAKOULUTUS },
    ],
  },
  { tyyppi: KOULUTUSTYYPPI.VAPAA_SIVISTYSTYO },
  { tyyppi: 'yleissivistava', children: [{ tyyppi: KOULUTUSTYYPPI.LUVA }] },
  { tyyppi: KOULUTUSTYYPPI.MUUT_KOULUTUKSET },
  { tyyppi: KOULUTUSTYYPPI.PERUSOPETUKSEN_LISAOPETUS },
];

export const TUTKINTOON_JOHTAVA_KOULUTUSTYYPPIHIERARKIA = [
  { tyyppi: KOULUTUSTYYPPI.AMMATILLINEN_KOULUTUS },
  {
    tyyppi: 'korkeakoulutus',
    children: [
      { tyyppi: KOULUTUSTYYPPI.YLIOPISTOKOULUTUS },
      { tyyppi: KOULUTUSTYYPPI.AMKKOULUTUS },
    ],
  },
  { tyyppi: KOULUTUSTYYPPI.LUKIOKOULUTUS },
];

export const KOULUTUSTYYPPI_TO_KOULUTUSTYYPPI_IDS_MAP = {
  [KOULUTUSTYYPPI.AMMATILLINEN_KOULUTUS]: [
    'koulutustyyppi_1',
    'koulutustyyppi_4',
    'koulutustyyppi_5',
    'koulutustyyppi_7',
    'koulutustyyppi_8',
    'koulutustyyppi_11',
    'koulutustyyppi_12',
    'koulutustyyppi_13',
    'koulutustyyppi_18',
    'koulutustyyppi_19',
    'koulutustyyppi_26',
  ],
  [KOULUTUSTYYPPI.KORKEAKOULUKOULUTUS]: ['koulutustyyppi_3'],
  [KOULUTUSTYYPPI.LUKIOKOULUTUS]: [
    'koulutustyyppi_2',
    'koulutustyyppi_9',
    'koulutustyyppi_14',
    'koulutustyyppi_23',
  ],
  [KOULUTUSTYYPPI.MUU_KOULUTUS]: [
    'koulutustyyppi_6',
    'koulutustyyppi_10',
    'koulutustyyppi_15',
    'koulutustyyppi_16',
    'koulutustyyppi_24',
    'koulutustyyppi_21',
    'koulutustyyppi_22',
  ],
  [KOULUTUSTYYPPI.YLIOPISTOKOULUTUS]: ['koulutustyyppi_3'],
  [KOULUTUSTYYPPI.AMKKOULUTUS]: ['koulutustyyppi_3'],
};

export const HAKULOMAKETYYPPI = {
  ATARU: 'ataru',
  HAKUAPP: 'haku-app',
  MUU: 'muu',
  EI_SAHKOISTA_HAKUA: 'ei sähköistä',
};

export const VALINTAPERUSTEET_KIELITAITO_MUU_OSOITUS_KOODI_URI =
  'kielitaidonosoittaminen_04';

export const POHJAVALINTA = {
  UUSI: 'uusi',
  KOPIO: 'kopio',
  AIEMPI: 'aiempi',
};

export const OPETUSHALLITUS_ORGANISAATIO_OID = '1.2.246.562.10.00000000001';

export const KOULUTUS_ROLE = 'APP_TARJONTA';

export const TOTEUTUS_ROLE = 'APP_TARJONTA';

export const HAKUKOHDE_ROLE = 'APP_TARJONTA';

export const HAKU_ROLE = 'APP_TARJONTA';

export const VALINTAPERUSTE_ROLE = 'APP_TARJONTA';

export const HAKUTAPA_YHTEISHAKU_KOODI_URI = 'hakutapa_01';

export const LIITTEEN_TOIMITUSTAPA = {
  TOIMITETAAN_LAHETTAMISEN_YHTEYDESSA: 'toimitetaan_lahettamisen_yhteydessa',
  JARJESTAJAN_OSOITE: 'jarjestajan_osoite',
  MUU_OSOITE: 'muu_osoite',
};
