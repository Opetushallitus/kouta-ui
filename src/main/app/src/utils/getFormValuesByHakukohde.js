import get from 'lodash/get';

import { isNumeric } from './index';
import getValintakoeFieldsValues from './getValintakoeFieldsValues';
import getHakulomakeFieldsValues from './getHakulomakeFieldsValues';

const getFormValuesByHakukohde = hakukohde => {
  const {
    alkamiskausiKoodiUri = '',
    kaytetaanHaunAikataulua,
    kielivalinta = [],
    minAloituspaikat = '',
    maxAloituspaikat = '',
    hakuajat = [],
    liitteetOnkoSamaToimitusaika,
    liitteetOnkoSamaToimitusosoite,
    liitteet = [],
    alkamisvuosi,
    liitteidenToimitusosoite = {},
    liitteidenToimitusaika,
    liitteidenToimitustapa,
    nimi = {},
    toinenAsteOnkoKaksoistutkinto,
    valintakokeet = [],
    pohjakoulutusvaatimusKoodiUrit = [],
    valintaperuste = '',
    minEnsikertalaisenAloituspaikat = '',
    maxEnsikertalaisenAloituspaikat = '',
    kaytetaanHaunHakulomaketta,
    hakulomaketyyppi,
    hakulomakeAtaruId,
    hakulomakeKuvaus,
    hakulomakeLinkki,
    kaytetaanHaunAlkamiskautta,
    tila,
  } = hakukohde;

  return {
    tila,
    alkamiskausi: {
      eriAlkamiskausi: !kaytetaanHaunAlkamiskautta,
      kausi: alkamiskausiKoodiUri,
      vuosi: {
        value: isNumeric(alkamisvuosi) ? alkamisvuosi.toString() : '',
      },
    },
    kieliversiot: kielivalinta,
    aloituspaikat: {
      minAloituspaikkamaara: isNumeric(minAloituspaikat)
        ? minAloituspaikat.toString()
        : '',
      maxAloituspaikkamaara: isNumeric(maxAloituspaikat)
        ? maxAloituspaikat.toString()
        : '',
      minEnsikertalaismaara: isNumeric(minEnsikertalaisenAloituspaikat)
        ? minEnsikertalaisenAloituspaikat.toString()
        : '',
      maxEnsikertalaismaara: isNumeric(maxEnsikertalaisenAloituspaikat)
        ? maxEnsikertalaisenAloituspaikat.toString()
        : '',
    },
    hakuajat: {
      eriHakuaika: !kaytetaanHaunAikataulua,
      hakuajat: (hakuajat || []).map(({ alkaa, paattyy }) => {
        return {
          alkaa: alkaa || '',
          paattyy: paattyy || '',
        };
      }),
    },
    perustiedot: {
      nimi,
      voiSuorittaaKaksoistutkinnon: !!toinenAsteOnkoKaksoistutkinto,
    },
    pohjakoulutus: (pohjakoulutusvaatimusKoodiUrit || []).map(value => ({
      value,
    })),
    valintaperusteenKuvaus: {
      value: valintaperuste,
    },
    valintakoe: getValintakoeFieldsValues(valintakokeet),
    liitteet: {
      toimitustapa: {
        tapa: liitteidenToimitustapa || '',
        paikka: {
          sahkoposti: get(liitteidenToimitusosoite, 'sahkoposti') || '',
          osoite: get(liitteidenToimitusosoite, 'osoite.osoite') || {},
          postinumero: get(
            liitteidenToimitusosoite,
            'osoite.postinumeroKoodiUri',
          )
            ? {
                value: liitteidenToimitusosoite.osoite.postinumeroKoodiUri,
              }
            : undefined,
        },
      },
      yhteinenToimituspaikka: Boolean(liitteetOnkoSamaToimitusosoite),
      yhteinenToimitusaika: Boolean(liitteetOnkoSamaToimitusaika),
      toimitusaika: liitteidenToimitusaika || '',
      liitteet: (liitteet || []).map(
        ({
          tyyppi,
          nimi = {},
          toimitusaika,
          toimitustapa,
          toimitusosoite,
          kuvaus = {},
        }) => {
          return {
            tyyppi: { value: tyyppi },
            nimi,
            kuvaus,
            toimitusaika: toimitusaika || '',
            toimitustapa: {
              tapa: toimitustapa || '',
              paikka: {
                osoite: get(toimitusosoite, 'osoite.osoite') || {},
                postinumero: get(toimitusosoite, 'osoite.postinumeroKoodiUri')
                  ? { value: toimitusosoite.osoite.postinumeroKoodiUri }
                  : undefined,
                sahkoposti: get(toimitusosoite, 'sahkoposti') || '',
              },
            },
          };
        },
      ),
    },
    hakulomake: {
      ...getHakulomakeFieldsValues({
        hakulomaketyyppi,
        hakulomakeAtaruId,
        hakulomakeKuvaus,
        hakulomakeLinkki,
      }),
      eriHakulomake: !kaytetaanHaunHakulomaketta,
    },
  };
};

export default getFormValuesByHakukohde;
