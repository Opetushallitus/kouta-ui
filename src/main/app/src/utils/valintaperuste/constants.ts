import { KOULUTUSTYYPPI } from '#/src/constants';

export const koulutustyypitWithValintatapa = [
  KOULUTUSTYYPPI.YLIOPISTOKOULUTUS,
  KOULUTUSTYYPPI.AMKKOULUTUS,
  KOULUTUSTYYPPI.AMMATILLINEN_OPETTAJAKOULUTUS,
  KOULUTUSTYYPPI.AMMATILLINEN_OPINTO_OHJAAJA_KOULUTUS,
  KOULUTUSTYYPPI.AMMATILLINEN_ERITYISOPETTAJA_KOULUTUS,
];
