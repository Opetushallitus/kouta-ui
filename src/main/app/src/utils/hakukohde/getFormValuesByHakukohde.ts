import _ from 'lodash';
import _fp from 'lodash/fp';

import { parseEditorState } from '#/src/components/Editor/utils';
import { LUKIO_YLEISLINJA } from '#/src/constants';
import { HakukohdeFormValues } from '#/src/types/hakukohdeTypes';
import { isNumeric, toSelectValue } from '#/src/utils';
import { getAjankohtaFields } from '#/src/utils/form/aloitusajankohtaHelpers';
import { getHakulomakeFieldsValues } from '#/src/utils/form/getHakulomakeFieldsValues';
import {
  getKokeetTaiLisanaytotValues,
  getTilaisuusValues,
} from '#/src/utils/form/getKokeetTaiLisanaytotValues';
import { mapValues } from '#/src/utils/lodashFpUncapped';

const getToimitustapaValues = (toimitustapa, toimitusosoite) => ({
  tapa: toimitustapa || '',
  paikka: {
    osoite: _fp.reduce(
      (acc, [kieli, osoite]) => {
        const [r1, r2] = `${osoite || ''}`.split('\n');
        acc.rivi1[kieli] = r1;
        acc.rivi2[kieli] = r2;
        return acc;
      },
      { rivi1: {}, rivi2: {} },
      Object.entries(toimitusosoite?.osoite?.osoite || {})
    ),
    postinumero: toimitusosoite?.osoite?.postinumeroKoodiUri
      ? { value: toimitusosoite?.osoite?.postinumeroKoodiUri }
      : undefined,
    sahkoposti: toimitusosoite?.sahkoposti || '',
    verkkosivu: toimitusosoite?.verkkosivu || '',
  },
});

// Keskiarvossa halutaan näyttää aina vähintään yksi desimaali (e.g. 7,0)
export const parseKeskiarvo = v =>
  v ? parseFloat(v).toLocaleString('fi', { minimumFractionDigits: 1 }) : '';

const getHakukohteenLinjaValues = ({
  linja,
  alinHyvaksyttyKeskiarvo,
  lisatietoa,
  painotetutArvosanat,
}) => ({
  linja: linja || LUKIO_YLEISLINJA,
  alinHyvaksyttyKeskiarvo: parseKeskiarvo(alinHyvaksyttyKeskiarvo),
  lisatietoa: mapValues(parseEditorState, lisatietoa),
  painotetutArvosanat: (painotetutArvosanat || []).map(arvosana => {
    return {
      painotettuOppiaine: {
        value: arvosana.koodiUrit?.oppiaine ?? '',
      },
      painokerroin: arvosana.painokerroin,
    };
  }),
});

export const getFormValuesByHakukohde = (
  hakukohde,
  formMode,
  nimiKoodista?: TranslatedField
): HakukohdeFormValues => {
  const {
    kaytetaanHaunAikataulua,
    kielivalinta = [],
    hakuajat = [],
    liitteetOnkoSamaToimitusaika,
    liitteetOnkoSamaToimitusosoite,
    liitteet = [],
    liitteidenToimitusosoite = {},
    liitteidenToimitusaika,
    liitteidenToimitustapa,
    nimi = {},
    hakukohdeKoodiUri,
    toinenAsteOnkoKaksoistutkinto,
    valintakokeet = [],
    pohjakoulutusvaatimusKoodiUrit = [],
    valintaperusteId,
    kaytetaanHaunHakulomaketta,
    hakulomaketyyppi,
    hakulomakeAtaruId,
    hakulomakeKuvaus,
    hakulomakeLinkki,
    jarjestyspaikkaOid,
    tila,
    esikatselu = false,
    pohjakoulutusvaatimusTarkenne,
    metadata = {},
    externalId,
    organisaatioOid,
  } = hakukohde;
  const {
    valintakokeidenYleiskuvaus,
    kynnysehto,
    kaytetaanHaunAlkamiskautta,
    koulutuksenAlkamiskausi = {},
    valintaperusteenValintakokeidenLisatilaisuudet = [],
    aloituspaikat,
    hakukohteenLinja,
    uudenOpiskelijanUrl,
  } = metadata;
  return {
    organisaatioOid: toSelectValue(organisaatioOid),
    externalId,
    uudenOpiskelijanUrl,
    tila,
    esikatselu,
    kieliversiot: kielivalinta,
    aloituspaikat: {
      aloituspaikkamaara: isNumeric(aloituspaikat?.lukumaara)
        ? aloituspaikat.lukumaara.toString()
        : '',
      ensikertalaismaara: isNumeric(aloituspaikat?.ensikertalaisille)
        ? aloituspaikat.ensikertalaisille.toString()
        : '',
      aloituspaikkakuvaus: mapValues(parseEditorState, aloituspaikat?.kuvaus),
    },
    hakuajat: {
      eriHakuaika: !kaytetaanHaunAikataulua,
      hakuajat: (hakuajat || []).map(({ alkaa = '', paattyy = '' }) => ({
        alkaa,
        paattyy,
      })),
    },
    perustiedot: {
      nimi: _fp.isEmpty(nimiKoodista) ? nimi : nimiKoodista,
      hakukohdeKoodiUri: toSelectValue(hakukohdeKoodiUri),
      voiSuorittaaKaksoistutkinnon: Boolean(toinenAsteOnkoKaksoistutkinto),
    },
    ajankohta: {
      kaytetaanHakukohteenAlkamiskautta:
        kaytetaanHaunAlkamiskautta === undefined
          ? false
          : !kaytetaanHaunAlkamiskautta,
      ...getAjankohtaFields(koulutuksenAlkamiskausi),
    },
    pohjakoulutus: {
      pohjakoulutusvaatimus: (pohjakoulutusvaatimusKoodiUrit || []).map(
        value => ({
          value,
        })
      ),
      tarkenne: _.mapValues(
        pohjakoulutusvaatimusTarkenne || {},
        parseEditorState
      ),
    },
    valintaperusteenKuvaus: {
      valintaperuste: valintaperusteId
        ? {
            value: valintaperusteId,
          }
        : undefined,
      kynnysehto: _.mapValues(kynnysehto, parseEditorState),
    },
    valintakokeet: {
      ...getKokeetTaiLisanaytotValues(
        valintakokeet,
        valintakokeidenYleiskuvaus,
        formMode
      ),
      // NOTE: tässä muutetaan taulukko [{id, tilaisuudet: [tilaisuus1, tilaisuus2]}] objektiksi {id: [tilaisuus1, tilaisuus2]} käsittelyn helpottamiseksi
      valintaperusteenValintakokeidenLisatilaisuudet: _fp.flow(
        _fp.keyBy('id'),
        _fp.mapValues(v => v.tilaisuudet.map(getTilaisuusValues))
      )(valintaperusteenValintakokeidenLisatilaisuudet),
    },
    jarjestyspaikkaOid,
    liitteet: {
      toimitustapa: getToimitustapaValues(
        liitteidenToimitustapa,
        liitteidenToimitusosoite
      ),
      yhteinenToimituspaikka: Boolean(liitteetOnkoSamaToimitusosoite),
      yhteinenToimitusaika: Boolean(liitteetOnkoSamaToimitusaika),
      toimitusaika: liitteidenToimitusaika || '',
      liitteet: (liitteet || []).map(
        ({
          tyyppiKoodiUri,
          nimi = {},
          toimitusaika,
          toimitustapa,
          toimitusosoite,
          kuvaus = {},
        }) => ({
          tyyppi: { value: tyyppiKoodiUri },
          nimi,
          kuvaus: _.mapValues(kuvaus, parseEditorState),
          toimitusaika: toimitusaika || '',
          toimitustapa: getToimitustapaValues(toimitustapa, toimitusosoite),
        })
      ),
    },
    hakulomake: {
      ...getHakulomakeFieldsValues({
        hakulomaketyyppi,
        hakulomakeAtaruId,
        hakulomakeKuvaus,
        hakulomakeLinkki,
      }),
      eriHakulomake: !kaytetaanHaunHakulomaketta,
    },
    hakukohteenLinja: hakukohteenLinja
      ? getHakukohteenLinjaValues(hakukohteenLinja)
      : undefined,
  };
};
