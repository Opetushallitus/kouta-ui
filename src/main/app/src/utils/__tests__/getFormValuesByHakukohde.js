import getFormValuesByHakukohde from '../getFormValuesByHakukohde';
import { HAKULOMAKETYYPPI } from '../../constants';

test('getFormValuesByHakukohde returns correct form values given hakukohde', () => {
  const values = getFormValuesByHakukohde({
    tila: 'tallennettu',
    alkamiskausiKoodiUri: 'kausi_1#1',
    kaytetaanHaunAikataulua: false,
    kielivalinta: ['fi', 'sv'],
    aloituspaikat: 35,
    hakuajat: [{ alkaa: '2019-03-29T12:28', paattyy: '2019-09-29T12:30' }],
    liitteetOnkoSamaToimitusaika: false,
    liitteetOnkoSamaToimitusosoite: false,
    liitteet: [
      {
        kuvaus: {
          fi: 'Fi kuvaus',
          sv: 'Sv kuvaus',
        },
        nimi: {
          fi: 'Fi nimi',
          sv: 'Sv nimi',
        },
        toimitusaika: '2016-10-25T09:45',
        toimitusosoite: {
          osoite: {
            osoite: {
              fi: 'Fi osoite',
              sv: 'Sv osoite',
            },
            postinumeroKoodiUri: 'posti_1#1',
            postitoimipaikka: {
              fi: 'Fi postitoimipaikka',
              sv: 'Sv postitoimipaikka',
            },
          },
          sahkoposti: {
            fi: 'Fi sahkoposti',
            sv: 'Sv sahkoposti',
          },
        },
        tyyppiKoodiUri: 'liitetyyppi_1#1',
        toimitustapa: 'muu_osoite',
      },
    ],
    alkamisvuosi: 2015,
    liitteidenToimitusosoite: {
      osoite: {
        osoite: {
          fi: 'Fi osoite',
          sv: 'Sv osoite',
        },
        postinumeroKoodiUri: 'posti_1#1',
        postitoimipaikka: {
          fi: 'Fi postitoimipaikka',
          sv: 'Sv postitoimipaikka',
        },
      },
      sahkoposti: {
        fi: 'Fi sahkoposti',
        sv: 'Sv sahkoposti',
      },
    },
    liitteidenToimitusaika: '2011-12-20T10:30',
    nimi: { fi: 'Fi nimi', sv: 'Sv nimi' },
    toinenAsteOnkoKaksoistutkinto: true,
    valintakokeet: [
      {
        tyyppiKoodiUri: 'tyyppi_1#1',
        tilaisuudet: [
          {
            osoite: {
              osoite: { fi: 'fi osoite', sv: 'sv osoite' },
              postinumeroKoodiUri: 'posti_1#1',
              postitoimipaikka: {
                fi: 'fi posititoimipaikka',
                sv: 'sv posititoimipaikka',
              },
            },
            aika: {
              alkaa: '2019-04-16T08:44',
              paattyy: '2019-04-18T08:44',
            },
            lisatietoja: {
              fi: 'fi lisatietoja',
              sv: 'sv lisatietoja',
            },
          },
        ],
      },
    ],
    pohjakoulutusvaatimusKoodiUrit: ['vaatimus_1#1', 'vaatimus_2#1'],
    pohjakoulutusvaatimusTarkenne: {
      fi: '<strong>Tarkenne fi</strong>',
      sv: '<strong>Tarkenne sv</strong>',
    },
    valintaperusteId: 'peruste_1#1',
    ensikertalaisenAloituspaikat: 39,
    hakulomakeAtaruId: '12345',
    hakulomaketyyppi: HAKULOMAKETYYPPI.ATARU,
    eriHakulomakeKuinHaulla: true,
    eriAlkamiskausi: true,
  });

  expect(values).toMatchSnapshot();
});
