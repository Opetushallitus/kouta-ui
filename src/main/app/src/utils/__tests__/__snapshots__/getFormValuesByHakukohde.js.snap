// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFormValuesByHakukohde returns correct form values given hakukohde 1`] = `
Object {
  "alkamiskausi": Object {
    "eriAlkamiskausi": true,
    "kausi": "kausi_1#1",
    "vuosi": Object {
      "value": "2015",
    },
  },
  "aloituspaikat": Object {
    "aloituspaikkamaara": "35",
    "ensikertalaismaara": "39",
  },
  "hakuajat": Object {
    "eriHakuaika": true,
    "hakuajat": Array [
      Object {
        "alkaa": "2019-03-29T12:28",
        "paattyy": "2019-09-29T12:30",
      },
    ],
  },
  "hakulomake": Object {
    "eriHakulomake": true,
    "kuvaus": Object {},
    "linkki": Object {},
    "lomake": Object {
      "value": "12345",
    },
    "tyyppi": "ataru",
  },
  "kieliversiot": Array [
    "fi",
    "sv",
  ],
  "liitteet": Object {
    "liitteet": Array [
      Object {
        "kuvaus": Object {
          "fi": "Fi kuvaus",
          "sv": "Sv kuvaus",
        },
        "nimi": Object {
          "fi": "Fi nimi",
          "sv": "Sv nimi",
        },
        "toimitusaika": "2016-10-25T09:45",
        "toimitustapa": Object {
          "paikka": Object {
            "osoite": Object {
              "fi": "Fi osoite",
              "sv": "Sv osoite",
            },
            "postinumero": Object {
              "value": "posti_1#1",
            },
            "sahkoposti": Object {
              "fi": "Fi sahkoposti",
              "sv": "Sv sahkoposti",
            },
          },
          "tapa": "muu_osoite",
        },
        "tyyppi": Object {
          "value": "liitetyyppi_1#1",
        },
      },
    ],
    "toimitusaika": "2011-12-20T10:30",
    "toimitustapa": Object {
      "paikka": Object {
        "osoite": Object {
          "fi": "Fi osoite",
          "sv": "Sv osoite",
        },
        "postinumero": Object {
          "value": "posti_1#1",
        },
        "sahkoposti": Object {
          "fi": "Fi sahkoposti",
          "sv": "Sv sahkoposti",
        },
      },
      "tapa": "",
    },
    "yhteinenToimitusaika": false,
    "yhteinenToimituspaikka": false,
  },
  "perustiedot": Object {
    "nimi": Object {
      "fi": "Fi nimi",
      "sv": "Sv nimi",
    },
    "voiSuorittaaKaksoistutkinnon": true,
  },
  "pohjakoulutus": Object {
    "pohjakoulutusvaatimus": Array [
      Object {
        "value": "vaatimus_1#1",
      },
      Object {
        "value": "vaatimus_2#1",
      },
    ],
    "tarkenne": Object {
      "fi":         "<p><strong>Tarkenne fi</strong></p>",
      "sv":         "<p><strong>Tarkenne sv</strong></p>",
    },
  },
  "tila": "tallennettu",
  "valintakoe": Object {
    "tilaisuudet": Object {
      "tyyppi_1#1": Array [
        Object {
          "alkaa": "2019-04-16T08:44",
          "lisatietoja": Object {
            "fi": "fi lisatietoja",
            "sv": "sv lisatietoja",
          },
          "osoite": Object {
            "fi": "fi osoite",
            "sv": "sv osoite",
          },
          "paattyy": "2019-04-18T08:44",
          "postinumero": Object {
            "value": "posti_1#1",
          },
        },
      ],
    },
    "tyypit": Array [
      Object {
        "value": "tyyppi_1#1",
      },
    ],
  },
  "valintaperusteenKuvaus": Object {
    "value": "peruste_1#1",
  },
}
`;
